lib/active_stash/search.rb

```diff
@@ -166,10 +165,15 @@ def query(*args, &block)
 
       # Reindex all records into CipherStash
       def reindex
-        find_each do |record|
-          record.save!(touch: false)
+        records = find_each.lazy.map do |r|
+          if r.stash_id.nil?
+            raise "Cannot index record ID=#{r.id} without a stash ID"
+          end
+          { id: r.stash_id, record: r.attributes }
```

Comment:
This code will only work for collections that have been indexex previously. The `save!` call in the old code would use ActiveRecord callbacks to ensure that a `stash_id` was set. This approach means a `stash_id` will _never_ be set.

Reply:
Well, actually (heh), this approach means that the reindex will fail, because it'll `raise` when it it hits the record that doesn't have a `stash_id`.  Calling `#save!` to set a stash ID won't work, because that'll also trigger an `upsert`, and that's not great because (a) an `upsert`-per-row is exactly what we're trying to avoid, and (b) we're in the middle of a `putStream` RPC just at the moment so trying to jam a `put` down the middle of that will end poorly.